<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1317</width>
    <height>820</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* Global dark canvas for Designer + runtime */
QWidget {
    background-color: #1e1e1e;
    color: #f0f0f0;
}

/* Inputs */
QLineEdit, QTextEdit, QPlainTextEdit, QSpinBox, QDoubleSpinBox, QComboBox {
    background-color: #2a2a2a;
    color: #f0f0f0;
    border: 1px solid #444;
    padding: 4px;
    selection-background-color: #3a3a3a;
}

/* Buttons */
QPushButton {
    background-color: #2d2d2d;
    border: 1px solid #444;
    padding: 6px 12px;
    border-radius: 6px;
}
QPushButton:hover { background-color: #353535; }
QPushButton:pressed { background-color: #2a2a2a; }

/* GroupBox */
QGroupBox {
    border: 1px solid #444;
    margin-top: 14px;
}
QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 6px;
}

/* Labels */
QLabel { color: #f0f0f0; }
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="lblTitle">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>0</y>
      <width>211</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>DSP Analyzer</string>
    </property>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>60</y>
      <width>331</width>
      <height>631</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="styleSheet">
        <string notr="true">QGroupBox {
    border: 2px solid gray;
    border-radius: 5px;
    margin-top: 10px;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px;
}
</string>
       </property>
       <property name="title">
        <string>GroupBox</string>
       </property>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>311</width>
          <height>201</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">QGroupBox {
    border: 2px solid gray;
    border-radius: 5px;
    margin-top: 10px;
	color:red;
	border-color:red;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px;
	color:red;

}
</string>
        </property>
        <property name="title">
         <string>GroupBox</string>
        </property>
        <widget class="QLabel" name="lblSampleRate">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>20</y>
           <width>221</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Örnekleme Frekanısını Girin (Hz):</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="txtSampleRate">
         <property name="geometry">
          <rect>
           <x>230</x>
           <y>20</y>
           <width>71</width>
           <height>26</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="lblDuration">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>50</y>
           <width>221</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Süreyi Girin (s):</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="txtDuration">
         <property name="geometry">
          <rect>
           <x>230</x>
           <y>50</y>
           <width>71</width>
           <height>26</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="lblSignalFreq">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>80</y>
           <width>231</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Sinyalin Frekans Değerini Girin(Hz):</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="txtSignalFreq">
         <property name="geometry">
          <rect>
           <x>250</x>
           <y>80</y>
           <width>51</width>
           <height>26</height>
          </rect>
         </property>
        </widget>
        <widget class="QPushButton" name="btnAddSignal">
         <property name="geometry">
          <rect>
           <x>200</x>
           <y>160</y>
           <width>101</width>
           <height>29</height>
          </rect>
         </property>
         <property name="text">
          <string>Sinyal Ekle</string>
         </property>
        </widget>
        <widget class="QComboBox" name="cmbSignalType">
         <property name="geometry">
          <rect>
           <x>200</x>
           <y>110</y>
           <width>101</width>
           <height>28</height>
          </rect>
         </property>
        </widget>
        <widget class="QDoubleSpinBox" name="dsbSignalAmplitude">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>160</y>
           <width>63</width>
           <height>29</height>
          </rect>
         </property>
         <property name="value">
          <double>1.000000000000000</double>
         </property>
        </widget>
        <widget class="QLabel" name="lblSignalType">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>110</y>
           <width>81</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Sinyal Türü:</string>
         </property>
        </widget>
        <widget class="QLabel" name="lblSignalType_2">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>160</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Genlik Değeri:</string>
         </property>
        </widget>
       </widget>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>240</y>
          <width>311</width>
          <height>81</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">QGroupBox {
    border: 2px solid gray;
    border-radius: 5px;
    margin-top: 10px;
	color:red;
	border-color:red;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px;
	color:red;

}
</string>
        </property>
        <property name="title">
         <string>GroupBox</string>
        </property>
        <widget class="QLabel" name="lblFFTScale">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>20</y>
           <width>181</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>FFT Ölçekleme Seçimi</string>
         </property>
        </widget>
        <widget class="QComboBox" name="cmbFFTScale">
         <property name="geometry">
          <rect>
           <x>170</x>
           <y>20</y>
           <width>131</width>
           <height>26</height>
          </rect>
         </property>
        </widget>
       </widget>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>320</y>
          <width>311</width>
          <height>108</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">QGroupBox {
    border: 2px solid gray;
    border-radius: 5px;
    margin-top: 10px;
	color:red;
	border-color:red;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px;
	color:red;

}
</string>
        </property>
        <property name="title">
         <string>GroupBox</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QLabel" name="lblCutoffBwSlider">
           <property name="text">
            <string>Cutoff / Bandwith Ayar Slider</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="lblSliderValue">
           <property name="text">
            <string>Değer: 5</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="sliderFilterParam">
           <property name="orientation">
            <enum>Qt::Orientation::Horizontal</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QGroupBox" name="groupBox_5">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>440</y>
          <width>311</width>
          <height>181</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">QGroupBox {
    border: 2px solid gray;
    border-radius: 5px;
    margin-top: 10px;
	color:red;
	border-color:red;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px;
	color:red;

}
</string>
        </property>
        <property name="title">
         <string>GroupBox</string>
        </property>
        <widget class="QLabel" name="lblWindowType">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>30</y>
           <width>171</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>FFT Penceresi Seçimi</string>
         </property>
        </widget>
        <widget class="QComboBox" name="cmbWindowType">
         <property name="geometry">
          <rect>
           <x>170</x>
           <y>30</y>
           <width>131</width>
           <height>26</height>
          </rect>
         </property>
        </widget>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>60</y>
      <width>461</width>
      <height>631</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QGroupBox" name="groupBox_6">
       <property name="title">
        <string>GroupBox</string>
       </property>
       <widget class="QGroupBox" name="groupBox_7">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>431</width>
          <height>181</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">QGroupBox {
    border: 2px solid gray;
    border-radius: 5px;
    margin-top: 10px;
	color:red;
	border-color:red;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px;
	color:red;

}
</string>
        </property>
        <property name="title">
         <string>GroupBox</string>
        </property>
        <widget class="QPushButton" name="btnLowPass">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>30</y>
           <width>111</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string>Low-Pass</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnHighPass">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>80</y>
           <width>111</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string>High-Pass</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnBandPass">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>130</y>
           <width>111</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string>Band-Pass</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnMovingAvg">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>130</y>
           <width>111</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string>Moving Avg </string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnBandStop">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>30</y>
           <width>111</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string>Band-Stop</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnAllPass">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>80</y>
           <width>111</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string>All-Pass</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnWiener">
         <property name="geometry">
          <rect>
           <x>290</x>
           <y>130</y>
           <width>111</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string>Wiener</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnKalman">
         <property name="geometry">
          <rect>
           <x>290</x>
           <y>80</y>
           <width>111</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string>Kalman</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnMedian">
         <property name="geometry">
          <rect>
           <x>290</x>
           <y>30</y>
           <width>111</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string>Median</string>
         </property>
        </widget>
       </widget>
       <widget class="QGroupBox" name="groupBox_8">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>200</y>
          <width>431</width>
          <height>211</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">QGroupBox {
    border: 2px solid gray;
    border-radius: 5px;
    margin-top: 10px;
	color:red;
	border-color:red;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px;
	color:red;

}
</string>
        </property>
        <property name="title">
         <string>GroupBox</string>
        </property>
        <widget class="QCustomPlot" name="customPlotTimeOriginal" native="true">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>20</y>
           <width>391</width>
           <height>181</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">border: 2px solid gray;
border-radius: 4px;
</string>
         </property>
        </widget>
       </widget>
       <widget class="QGroupBox" name="groupBox_9">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>410</y>
          <width>431</width>
          <height>211</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">QGroupBox {
    border: 2px solid gray;
    border-radius: 5px;
    margin-top: 10px;
	color:red;
	border-color:red;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px;
	color:red;

}
</string>
        </property>
        <property name="title">
         <string>GroupBox</string>
        </property>
        <widget class="QCustomPlot" name="customPlotTimeFiltered" native="true">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>20</y>
           <width>391</width>
           <height>181</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">border: 2px solid gray;
border-radius: 4px;
</string>
         </property>
        </widget>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>699</y>
      <width>1271</width>
      <height>51</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QGroupBox" name="groupBox_14">
       <property name="title">
        <string>GroupBox</string>
       </property>
       <widget class="QPushButton" name="btnAddNoise">
        <property name="geometry">
         <rect>
          <x>640</x>
          <y>20</y>
          <width>121</width>
          <height>29</height>
         </rect>
        </property>
        <property name="text">
         <string>Gürültü Ekle</string>
        </property>
       </widget>
       <widget class="QPushButton" name="btnSave">
        <property name="geometry">
         <rect>
          <x>1110</x>
          <y>20</y>
          <width>121</width>
          <height>29</height>
         </rect>
        </property>
        <property name="text">
         <string>Kaydet</string>
        </property>
       </widget>
       <widget class="QComboBox" name="cmbNoiseType">
        <property name="geometry">
         <rect>
          <x>180</x>
          <y>20</y>
          <width>131</width>
          <height>26</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="lblNoiseType">
        <property name="geometry">
         <rect>
          <x>40</x>
          <y>20</y>
          <width>131</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Gürültü Türü Seç:</string>
        </property>
       </widget>
       <widget class="QLabel" name="lblNoiseAmplitude">
        <property name="geometry">
         <rect>
          <x>350</x>
          <y>20</y>
          <width>131</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Gürültü Şiddeti Seç:</string>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="dsbNoiseAmplitude">
        <property name="geometry">
         <rect>
          <x>490</x>
          <y>20</y>
          <width>63</width>
          <height>29</height>
         </rect>
        </property>
        <property name="value">
         <double>1.000000000000000</double>
        </property>
       </widget>
       <widget class="QPushButton" name="btnClear">
        <property name="geometry">
         <rect>
          <x>970</x>
          <y>20</y>
          <width>121</width>
          <height>29</height>
         </rect>
        </property>
        <property name="text">
         <string>Temizle</string>
        </property>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>60</y>
      <width>461</width>
      <height>631</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QGroupBox" name="groupBox_10">
       <property name="title">
        <string>GroupBox</string>
       </property>
       <widget class="QGroupBox" name="groupBox_11">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>431</width>
          <height>181</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">QGroupBox {
    border: 2px solid gray;
    border-radius: 5px;
    margin-top: 10px;
	color:red;
	border-color:red;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px;
	color:red;

}
</string>
        </property>
        <property name="title">
         <string>GroupBox</string>
        </property>
        <widget class="QLabel" name="lblSampleRateValue">
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>50</y>
           <width>191</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Örnekleme Frekansı (Hz):</string>
         </property>
        </widget>
        <widget class="QLabel" name="lblDurationValue">
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>80</y>
           <width>221</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Süre (s):</string>
         </property>
        </widget>
        <widget class="QLabel" name="lblSignalFreqValue">
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>110</y>
           <width>231</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Sinyallerin Frekans Değerleri (Hz):</string>
         </property>
        </widget>
        <widget class="QLabel" name="txtSampleValue">
         <property name="geometry">
          <rect>
           <x>270</x>
           <y>50</y>
           <width>101</width>
           <height>20</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">QLabel {
    border: 2px solid gray;
    border-radius: 5px;
}</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QLabel" name="txtDurationValue">
         <property name="geometry">
          <rect>
           <x>270</x>
           <y>80</y>
           <width>101</width>
           <height>20</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">QLabel {
    border: 2px solid gray;
    border-radius: 5px;
}</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QLabel" name="txtSignalFreqValue">
         <property name="geometry">
          <rect>
           <x>270</x>
           <y>110</y>
           <width>101</width>
           <height>20</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">QLabel {
    border: 2px solid gray;
    border-radius: 5px;
}</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </widget>
       <widget class="QGroupBox" name="groupBox_12">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>200</y>
          <width>431</width>
          <height>211</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">QGroupBox {
    border: 2px solid gray;
    border-radius: 5px;
    margin-top: 10px;
	color:red;
	border-color:red;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px;
	color:red;

}
</string>
        </property>
        <property name="title">
         <string>GroupBox</string>
        </property>
        <widget class="QCustomPlot" name="customPlotFreqOriginal" native="true">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>20</y>
           <width>391</width>
           <height>181</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">border: 2px solid gray;
border-radius: 4px;
</string>
         </property>
        </widget>
       </widget>
       <widget class="QGroupBox" name="groupBox_13">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>410</y>
          <width>431</width>
          <height>211</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">QGroupBox {
    border: 2px solid gray;
    border-radius: 5px;
    margin-top: 10px;
	color:red;
	border-color:red;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px;
	color:red;

}
</string>
        </property>
        <property name="title">
         <string>GroupBox</string>
        </property>
        <widget class="QCustomPlot" name="customPlotFreqFiltered" native="true">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>20</y>
           <width>391</width>
           <height>181</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">border: 2px solid gray;
border-radius: 4px;
</string>
         </property>
        </widget>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1317</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
